src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingReconciliationService.kt

```diff
@@ -31,18 +31,24 @@ class SentencingReconciliationService(
     val remandAdjustmentTypes = listOf("RSR", "RX")
   }
 
-  suspend fun generateReconciliationReport(activePrisonersCount: Long): List<MismatchSentencingAdjustments> {
-    return activePrisonersCount.asPages(pageSize).flatMap { page ->
-      val activePrisoners = getActivePrisonersForPage(page)
+  suspend fun generateReconciliationReport(allPrisoners: Boolean, activePrisonersCount: Long): List<MismatchSentencingAdjustments> = activePrisonersCount.asPages(pageSize).flatMap { page ->
+    val activePrisoners = getActivePrisonersForPage(allPrisoners, page)
 
-      withContext(Dispatchers.Unconfined) {
-        activePrisoners.map { async { checkBookingAdjustmentsMatch(it) } }
-      }.awaitAll().filterNotNull()
-    }
+    withContext(Dispatchers.Unconfined) {
+      activePrisoners.map { async { checkBookingAdjustmentsMatch(it) } }
+    }.awaitAll().filterNotNull()
   }
 
-  private suspend fun getActivePrisonersForPage(page: Pair<Long, Long>) =
-    runCatching { doApiCallWithRetries { nomisApiService.getActivePrisoners(page.first, page.second) }.content }
+  // TODO - feature switch getActivePrisoners get all prisoners
+  private suspend fun getActivePrisonersForPage(@Suppress("UNUSED_PARAMETER") allPrisoners: Boolean, page: Pair<Long, Long>) =
+    runCatching {
+      doApiCallWithRetries {
```

Comment:
nomisApiService.getActivePrisoners has a 'retryWhen' which means doApiCallWithRetries shouldnt be needed (though it only retries network-timeout type errors rather than any errors)

Reply:
Thanks